### PR TITLE
docs(datasource/custom): rewrite debugging section

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Write short sentences
- Use common words
- Use ordered list
- Add headings
- Use active voice and present tense
- _Emphasize_ the key difference in one sentence
- Add a good title to the JSON codeblock
- Link to the docs for the configuration options
- Explain who the information is for with the `If you ... then do ...` style sentences

## Context

Rewrite the text that `@secustor` added with:

- #30241

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/custom/readme.md [hunk 1]
```diff
@@ -83,10 +83,14 @@ All available options:
...
 
-Renovate writes tracing log entries before transformation starts and after if Renovate finds an unexpected data format.
-To surface this data you can use `logLevelRemap` to surface this data on the App or run Renovate in dryRun mode and log level set to `trace` locally.
+Renovate writes tracing logs entries before transformation.
+If Renovate finds an unexpected dataformat, it also writes a tracing log _after_ transformation.
 
-```json
+#### Getting trace level logs on hosted app
+
+If you use the Mend Renovate app, use the [`logLevelRemap` config option](https://docs.renovatebot.com/configuration-options/#loglevelremap) to get the trace log.
```
Comment (HonkingGoose) [inline lib/modules/datasource/custom/readme.md:91]: Please:

- check if this is the correct config option
- suggest the correct link format

File: lib/modules/datasource/custom/readme.md [hunk 2]
```diff
@@ -97,6 +101,13 @@ To surface this data you can use `logLevelRemap` to surface this data on the App
 }
 ```
 
+#### Getting trace level logs when self-hosting
+
+If you self-host Renovate, follow these steps to get the trace logs:
+
+1. Set the [`logFileLevel` config option](https://docs.renovatebot.com/self-hosted-configuration/#logfilelevel) to `trace`
+1. Run Renovate in `[dryRun` mode](https://docs.renovatebot.com/self-hosted-configuration/#dryrun)
```
Comment (HonkingGoose) [inline lib/modules/datasource/custom/readme.md:109]: Same thing as above, please check correct name and suggest correct link format.

Comment (secustor) [inline lib/modules/datasource/custom/readme.md:109]:
```suggestion
1. Run Renovate in [`dryRun` mode](../../../self-hosted-configuration.md#dryrun)
```

File: lib/modules/datasource/custom/readme.md [hunk 3]
```diff
@@ -97,6 +101,13 @@ To surface this data you can use `logLevelRemap` to surface this data on the App
 }
 ```
 
+#### Getting trace level logs when self-hosting
+
+If you self-host Renovate, follow these steps to get the trace logs:
+
+1. Set the [`logFileLevel` config option](https://docs.renovatebot.com/self-hosted-configuration/#logfilelevel) to `trace`
```
Comment (secustor) [inline lib/modules/datasource/custom/readme.md:108]:
```suggestion
1. Set the [`logFileLevel` config option](../../../self-hosted-configuration.md#logfilelevel) to `trace`
```

File: lib/modules/datasource/custom/readme.md (hunk 1 above)
Comment (secustor) [inline lib/modules/datasource/custom/readme.md:91]:
```suggestion
If you use the Mend Renovate app, use the [`logLevelRemap` config option](../../../self-hosted-configuration.md#loglevelremap) to get the trace log.
```

**Review by secustor — COMMENTED** [review]
Thanks for rewriting the docs. I have added the relative links and a fix for the linting issue.

File: lib/modules/datasource/custom/readme.md (hunk 1 above)
Comment (HonkingGoose) [reply lib/modules/datasource/custom/readme.md:91]: The published docs have `logLevelRemap` as a _repository_ config option. So `self-hosted-configuration.md` is the wrong file, I think.

I'm skipping this suggestion. But we should still put the right link format in! :wink:

Comment (secustor) [reply lib/modules/datasource/custom/readme.md:91]: You are right. Copy and paste error of mine.

**Review by rarkins — APPROVED** [review]

**Pull request merged**